aiohttp/client_proto.py

```diff
@@ -80,41 +82,67 @@ def is_connected(self) -> bool:
     def connection_lost(self, exc: Optional[BaseException]) -> None:
         self._drop_timeout()
 
-        if exc is not None:
-            set_exception(self.closed, exc)
-        else:
+        original_connection_error = exc
+        reraised_exc = original_connection_error
+
+        connection_closed_cleanly = original_connection_error is None
+        """Whether the connection got a clean EOF."""
```

Comment:
Should this be a regular comment? Seems odd having a multiline str.

Reply:
It's a variant of an attribute docstring for the variable above ([PEP 258](https://peps.python.org/pep-0258/#attribute-docstrings)). Sphinx recognizes these in some contexts. I like using these to explain the data, as in "_what_ a thing represents". The comments, OTOH, have different semantics — they are typically used to explain the unobvious motivation, as in "_why_" (although they are indeed often confused with docstrings).

Reply:
But maybe a shorter comment would be fine too. I'll think about it.

Reply:
I hope the variable name conveys the semantic meaning well enough. But it's unobvious that `None` is a sentinel for "the protocol got EOF as expected".

Reply:
Right, although that PEP is reject, and upon reading it, this isn't a valid attribute docstring as defined there either:

> The assignment must be in one of the following contexts:
> 
>     At the top level of a module (i.e., not nested inside a compound statement such as a loop or conditional): a module attribute.
>     At the top level of a class definition: a class attribute.
>     At the top level of the “__init__” method definition of a class: an instance attribute. Instance attributes assigned in other methods are assumed to be implementation details. (@@@ __new__ methods?)
>     A function attribute assignment at the top level of a module or class definition.

Reply:
> I hope the variable name conveys the semantic meaning well enough

Yes, I'd say the variable name is pretty clear, probably don't need the comment at all.

Reply:
```suggestion
```